components/camera/videosource/webcam.go

```diff
@@ -532,51 +570,64 @@ func (c *webcam) startBuffer() {
 			case <-closedCtx.Done():
 				return
 			case <-ticker.C:
-				// We must unlock the mutex even if the release() or read() functions panic.
-				func() {
-					c.mu.Lock()
-					defer c.mu.Unlock()
-					if c.buffer.release != nil {
-						c.buffer.release()
-						c.buffer.release = nil
-						c.buffer.frame = nil
+				// Make a private copy of the previously published frame so consumers can continue to read it,
+				// then return the Reader-managed buffer before we kick off the next read. This avoids holding
+				// on to driver memory while still serving the last frame.
```

Comment:
Introducing `prevRelease` fixes the hang. All other changes on this PR are to supplement it, to ensure best practices, and fix vulnerabilities in the code I saw along the way of fixing the hang.